src/openforms/validations/tests/test_registry.py

```diff
@@ -2,43 +2,66 @@
 from django.test import TestCase
 from django.utils.translation import gettext as _
 
+from rest_framework import serializers
 from rest_framework.exceptions import ValidationError as DRFValidationError
 
-from openforms.validations.registry import Registry
+from openforms.validations.registry import Registry, StringValueSerializer
+
+
+class DummyDataSerializer(serializers.Serializer):
+    field = serializers.CharField()
+
+
+class DictValueSerializer(serializers.Serializer):
+    value = DummyDataSerializer()
 
 
 class DjangoValidator:
     is_enabled = True
     components = ("textfield",)
+    value_serializer = StringValueSerializer
 
-    def __call__(self, value):
+    def __call__(self, value, submission):
         if value != "VALID":
             raise DjangoValidationError("not VALID value")
 
 
 class DRFValidator:
     is_enabled = True
     components = ("phoneNumber",)
+    value_serializer = StringValueSerializer
 
-    def __call__(self, value):
+    def __call__(self, value, submission):
         if value != "VALID":
             raise DRFValidationError("not VALID value")
 
 
 class DisabledValidator:
     is_enabled = False
     components = ("textfield",)
+    value_serializer = StringValueSerializer
 
-    def __call__(self, value):
+    def __call__(self, value, submission):
         if value != "VALID":
             raise DRFValidationError("not VALID value")
 
 
-def function_validator(value):
+class DictValidator:
+    is_enabled = True
+    value_serializer = DictValueSerializer
+
+    def __call__(self, value, submission):
+        return True
+
+
+def function_validator(value, submission):
     if value != "VALID":
         raise DjangoValidationError("not VALID value")
 
 
+function_validator.value_serializer = StringValueSerializer
```

Comment:
oh this only concerns tests to set the attribute? If we don't have any actual function-based validator implementations, just drop support for it.

Reply:
I think the next refactor PR handles this